File: OpenWeatherAPI/OpenWeatherAPI.Models/Models/WeatherConditionDTO.cs
using OpenWeatherAPI.Business.Base;
using OpenWeatherAPI.Models.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace OpenWeatherAPI.Business.Models
{
    public class WeatherConditionDTO : BaseEntity
    {
        public string Base { get; set; }
        public int Pressure { get; set; }
        public int Humidity { get; set; }
        public int? SeaLevel { get; set; }
        public int? GroundLevel { get; set; }
        public double WindSpeed { get; set; }
        public int WindDegrees { get; set; }
        public double? WindGust { get; set; }
        public int Clouds { get; set; }
        public double? RainVolume1h { get; set; }
        public double? RainVolume3h { get; set; }
        public double? SnowVolume1h { get; set; }
        public double? SnowVolume3h { get; set; }
        public int Sunrise { get; set; }
        public int Sunset { get; set; }
        public int Timezone { get; set; }
        public int Dt { get; set; }
        public int CityId { get; set; }
        public int StatusCode { get; set; }
        public DateTime RegDate { get; set; }
        public UnitsDTO Units { get; set; }
        public CityDTO City { get; set; }
        public ICollection<WeatherTypeDTO> WeatherType { get; set; }
    }
}

File: OpenWeatherWebApp/src/app/shared/DTOs/WeatherConditionDTO.ts
import { CityDTO } from "./CityDTO";
import { UnitsDTO } from "./UnitsDTO";
import { WeatherTypeDTO } from "./WeatherTypeDTO";

export class WeatherConditionDTO
{
    constructor(
        public id:number,
        public base:string,
        public pressure:number,
        public humidity:number,
        public seaLevel:number,
        public groundLevel:number,
        public windSpeed:number,
        public windDegrees:number,
        public windGust:number,
        public clouds:number,
        public rainVolume1h:number,
        public rainVolume3h:number,
        public snowVolume1h:number,
        public snowVolume3h:number,
        public sunrise:number,
        public sunset:number,
        public timezone:number,
        public dt:number,
        public cityId:number,
        public statusCode:number,
        public regDate:Date,
        public units:UnitsDTO,
        public city:CityDTO,
        public weatherType:WeatherTypeDTO[]
    ){}
}